dbt_platform_helper/providers/ecr.py

```diff
@@ -18,3 +23,55 @@ def get_ecr_repo_names(self) -> list[str]:
         for page in self._get_client().get_paginator("describe_repositories").paginate():
             out.extend([repo["repositoryName"] for repo in page.get("repositories", {})])
         return out
+
+    def get_image_details(
+        self, application: Application, codebase: str, image_ref: str
+    ) -> list[dict]:
+        """Check if image exists in AWS ECR, and return a list of dictionaries
+        containing image metadata."""
+
+        repository = f"{application.name}/{codebase}"
+
+        try:
+            image_info = self._get_client().describe_images(
+                repositoryName=repository,
+                imageIds=[{"imageTag": image_ref}],
+            )
+
+            self._check_image_details_exists(image_info, image_ref)
+
+            return image_info.get("imageDetails")
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "ImageNotFoundException":
+                raise ImageNotFoundException(image_ref)
+            if e.response["Error"]["Code"] == "RepositoryNotFoundException":
+                raise RepositoryNotFoundException(repository)
+
+    @staticmethod
+    def _check_image_details_exists(image_info: dict, image_ref: str):
+        """Error handling for any unexpected scenario where AWS ECR returns a
+        malformed response."""
+
+        if "imageDetails" not in image_info:
+            raise ImageNotFoundException(image_ref)
```

Comment:
Might be nice to put this "private" method under the public ones.

Reply:
Not sure what you mean? This is already in a private method 🤔

Reply:
Oh I see. Doesn't isort just auto sort everything though? I'll try it.